lib/spack/docs/repositories.rst

```diff
@@ -0,0 +1,275 @@
+  but that support is not implemented yet.
+
+---------------------
+Namespaces
+---------------------
+
+Every repository in Spack has an associated **namespace** defined in its
+top-level ``repo.yaml`` file.  The namespace serves only to distinguish
+packages from different repositories.  If you look at the contents of
+``var/spack/repos/builtin/repo.yaml`` in the builtin repository, you'll
+see that its namespace is ``builtin``:
+
+.. code-block:: console
+
+  $ cat var/spack/repos/builtin/repo.yaml
+  repo:
+    namespace: builtin
+
+The namespace doesn't **have** to correspond to the repository directory
+name; it's just convenient in this case.
+
+If you make a repository for packages from your organization, you might
+use your organization's name.  You can also nest namespaces using
+periods.  For example, LLNL might use a namespace for its internal
+repositories like ``llnl``. Packages from the Physical & Life Sciences
+directorate (PLS) might use the ``llnl.pls`` namespace, and packages
+created by the computation directorate might use ``llnl.comp``.
+
+You can have packages with the same name from different namespaces
+installed at once.  For example, LLNL might maintain its own version of
+``mpich`` separate from Spack's builtin ``mpich`` package.  If you just
+use ``spack find``, you won't see a difference between these two
+packages:
+
+.. code-block:: console
+
+  $ spack find
+  ==> 2 installed packages.
+  -- linux-rhel6-x86_64 / gcc@4.4.7 -------------
+  mpich@3.2  mpich@3.2
+
+However, if you use ``spack find -N``, Spack will display the packages
+with their namespaces:
+
+.. code-block:: console
+
+  $ spack find -N
+  ==> 2 installed packages.
+  -- linux-rhel6-x86_64 / gcc@4.4.7 -------------
+  builtin.mpich@3.2  llnl.comp.mpich@3.2
+
+As you might guess, packages that are identical save for their namespace
+will still have different hashes:
+
+.. code-block:: console
+
+  $ spack find -lN
+  ==> 2 installed packages.
+  -- linux-rhel6-x86_64 / gcc@4.4.7 -------------
+  c35p3gc builtin.mpich@3.2  itoqmox llnl.comp.mpich@3.2
+
+All Spack commands that take a package :ref:`spec <sec-specs>` can also
+accept a qualified spec with a namespace, so you can use the namespace to
+be more specific when designating, e.g., which package to uninstall:
+
+.. code-block:: console
+
+  spack uninstall llnl.comp.mpich
+
+Or, if you have your own repository registered and you want to **force**
+spack to use the ``builtin`` implementation of ``mpich`` for a certain
+build of ``hdf5``, you could do this:
+
+.. code-block:: console
+
+  spack install hdf5 +mpi ^llnl.comp.mpich
+
+---------------------
+Listing repositories
+---------------------
+
+Spack's :ref:`configuration system <configuration>` allows repository
+settings to come from ``repos.yaml`` files in many locations.  If you
+want to see the repositories registered as a result of all configuration
+files, use ``spack repo list``.
+
+.. _cmd-spack-repo-list:
```

Comment:
We established an unofficial convention that the [Command Index](http://spack.readthedocs.io/en/latest/command_index.html) should contain subcommands (`spack compiler`, `spack repo`) but not subsubcommands (`spack compiler add`, `spack compiler find`, `spack compiler remove`, ...) since they are always documented in the same area.